src/Cargo.toml

```diff
@@ -20,3 +20,11 @@ members = [
     "lib/vasi-macro",
     "lib/vasi-sync",
 ]
+
+[profile.dev]
+# shim/src/tls.rs, and maybe other places, needs this
+# to avoid smashing the stack with a large immediate temporary.
+# opt-level 1 is enough to optimize away the offending temporary.
+# 
+# See https://stackoverflow.com/questions/25805174/creating-a-fixed-size-array-on-heap-in-rust/68122278#68122278
+opt-level = 1 
```

Comment:
I think this will also affect things like stack traces in debug builds, which might make it more difficult to debug issues. Which part of the code is overflowing the stack? I only see things in the code that look like they're at most 24 KiB, which I think wouldn't cause a problem? Are there any other options, like requiring the type to be `Copy`, passing references around, and using `ptr.write()`? Or constructing it in a `const` context so that it gets embedded into the binary instead?

Reply:
Why is the stack so small (16 KB)?

Reply:
> Which part of the code is overflowing the stack?

I don't know of a good way to identify for sure, other than trial and error or poking through assembly/gdb looking for `rsp` decrements. https://crates.io/crates/cargo-call-stack looked promising, but I couldn't get it working.

> Why is the stack so small (16 KB)?

This test uses a small stack mostly to push how small of a stack we can run on; ideally it'd be a separate test I suppose.

As mentioned in the PR description, `MINSIGSTKSZ` and `SIGSTKSZ` (described in `sigaltstack(2)`) are even smaller, so it'd be nice not to require a stack a lot larger than that.

I'd rather try things this way for now, and if the opt level makes debugging too onerous falling back to increasing this test stack size.

Reply:
Maybe add a comment to the `opt-level = 1` to point it at this test, if it's the only place we use a stack small enough to run into this issue?

Reply:
done